### PR TITLE
fix pricing

### PR DESCRIPTION
**Pull request by derkgort**
fix pricing so it will get all prices in each zone instead of only the firewall.

Also added an option to return the array as associative array to make it easier to get to a zone.
Also made an option in Server to use that associative array, but made sure it will not change current behaviour.

### CODE REVIEW

**Comment by derkgort** [comment]
I'll fix them

**Comment by derkgort** [comment]
Yeah. This one is just being a testing tool. The API property name, and with camelCasing this is what the variable need to be called.
I could name remove the public part, but that's just changing it because some tool says so..

File: src/Entity/Pricing.php [hunk 1]
```diff
@@ -17,100 +16,101 @@
...
-
-            // Prevent from double processing.
-            unset($parameters[$property]);
+            $ccProperty = static::convertToCamelCase($property);
+            $this->$ccProperty = $pricingItem;
         }
 
         parent::build($parameters);
     }
+
```
Comment (malc0mn) [inline src/Entity/Pricing.php]: This line and the line below are not needed I guess ;-)

File: src/Api/AbstractApi.php [hunk 2]
```diff
@@ -30,4 +30,20 @@ public function __construct(AdapterInterface $adapter, $endpoint = null)
...
     }
+
+    /**
+     * Return new assoc array that uses $propertyForKey as key.
+     *
+     * @param array  $array
+     * @param string $propertyForKey
+     * @return array
+     */
+    protected function array_map_keys(array $array, $propertyForKey) {
```
Comment (malc0mn) [inline src/Api/AbstractApi.php]: Can we rename this method to camelCase plz.? Also, move the opening curly to the next line (PSR-1/2)...

**Comment by malc0mn** [comment]
Thanks for the contribution! I spotted some PSR coding standard issues, if you could address those, that'd be great!

**Comment by derkgort** [comment]
Just updated them.

**Pull request merged**